hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/IncrSourceHelper.java

```diff
@@ -107,25 +112,30 @@ public static QueryInfo generateQueryInfo(JavaSparkContext jssc, String srcBaseP
         .setConf(HadoopFSUtils.getStorageConfWithCopy(jssc.hadoopConfiguration()))
         .setBasePath(srcBasePath).setLoadActiveTimelineOnLoad(true).build();
 
+    // TODO(yihua): handle transition time in CheckpointV1
     HoodieTimeline completedCommitTimeline = srcMetaClient.getCommitsAndCompactionTimeline().filterCompletedInstants();
     final HoodieTimeline activeCommitTimeline = handleHollowCommitIfNeeded(completedCommitTimeline, srcMetaClient, handlingMode);
     Function<HoodieInstant, String> timestampForLastInstant = instant -> handlingMode == HollowCommitHandling.USE_TRANSITION_TIME
         ? instant.getCompletionTime() : instant.requestedTime();
-    String beginInstantTime = beginInstant.orElseGet(() -> {
+
+    Option<Checkpoint> translatedCheckpoint = beginInstant.isPresent()
```

Comment:
this is for the code 1.0/tableVersion=6 path.. where it can continue to write old key? IIUC this is orthogonal to using transition time etc..